src/module.ts

```diff
@@ -1081,14 +1081,100 @@ export const enum RefAsOp {
 export const enum BrOnOp {
   /** br_on_null */
   BrOnNull = 0 /* TODO_BinaryenBrOnNull */,
+  /** br_on_non_null */
+  BrOnNonNull = 1 /* TODO_BinaryenBrOnNonNull */,
   /** br_on_cast */
-  BrOnCast = 1 /* TODO_BinaryenBrOnCast */,
+  BrOnCast = 2 /* TODO_BinaryenBrOnCast */,
+  /** br_on_cast_fail */
+  BrOnCastFail = 3 /* TODO_BinaryenBrOnCastFail */,
   /** br_on_func */
-  BrOnFunc = 2 /* TODO_BinaryenBrOnFunc */,
+  BrOnFunc = 4 /* TODO_BinaryenBrOnFunc */,
+  /** br_on_non_func */
+  BrOnNonFunc = 5 /* TODO_BinaryenBrOnNonFunc */,
   /** br_on_data */
-  BrOnData = 3 /* TODO_BinaryenBrOnData */,
+  BrOnData = 6 /* TODO_BinaryenBrOnData */,
+  /** br_on_non_data */
+  BrOnNonData = 7 /* TODO_BinaryenBrOnNonData */,
   /** br_on_i31 */
-  BrOnI31 = 4 /* TODO_BinaryenBrOnI31 */
+  BrOnI31 = 8 /* TODO_BinaryenBrOnI31 */,
+  /** br_on_non_i32 */
+  BrOnNonI31 = 9 /* TODO_BinaryenBrOnNonI31 */
```

Comment:
Also wdyt refactor all sych ops from `BrOnOp.BrOnFunc` to `BrOnOp.Func` Due to `BrOnOp.BrOn...` and etc looks redundant

Reply:
Didn't do any renaming yet, but yeah, these renamings make sense to me :)

Reply:
Yeah. Better do this in separate PR

Reply:
Did this here for the new ones not referenced otherwise now, but there's also `RefIsOp` and `RefAsOp` to update later.